### DISCUSSION TITLE
Adding Items to NativeMenuBar at Runtime

### DISCUSSION THREAD

**Discussion by samuel-roberts** [AvaloniaUI/Avalonia#5956] [Q&A] [answered]
I would like to add a menu to my `NativeMenuBar` from C# code. Below is a simplified snipped of my .axaml file:

```xml
<Window Name="MainWindow">
    <NativeMenu.Menu>
        <NativeMenu>
            <NativeMenuItem Header="File">
                <NativeMenuItem.Menu>
                    <NativeMenu>
                        <NativeMenuItem Header="New" Command="{Binding #MainWindow.New}" Gesture="Ctrl+N" />
                        <NativeMenuItem Header="Open" Command="{Binding #MainWindow.Open}" Gesture="Ctrl+O" />
                        <NativeMenuItem Header="Save" Command="{Binding #MainWindow.Save}" Gesture="Ctrl+S" />
                        <NativeMenuItem Header="Save As" Command="{Binding #MainWindow.SaveAs}" />
                    </NativeMenu>
                </NativeMenuItem.Menu>
            </NativeMenuItem>
          </NativeMenu>
    </NativeMenu.Menu>
     
    <DockPanel>
        <NativeMenuBar DockPanel.Dock="Top" x:Name="MenuBar" />
          
            <!-- Rest of code -->
          
    <DockPanel>
</Window>
```

I'd like to add a new menu alongside "File" containing some options that are populated at runtime. I can find the control with `this.FindControl<NativeMenuBar>("MenuBar");` but I have yet to figure out a way of adding items to it. Is it (currently) possible to do this? And if so, how?

Thank you.

**Comment by maxkatz6** [comment, marked as answer]
What specific issue do you have with updating this menu in runtime? I remember I did it in 0.10.0 just by updating NativeMenu.Items array, and it worked without problems.

**Reply by samuel-roberts** [reply]
Ah ok, then I guess my question is how do I access the `NativeMenu` object from the code behind? I can't seem to access it with `Find<NativeMenu>()` as I can't attach an `x:Name` to the menu

**Reply by maxkatz6** [reply]
You can get native menu for from the window object

**Reply by maxkatz6** [reply]
NativeMenu.GetMenu(window) or something

**Reply by samuel-roberts** [reply]
Perfect! Thank you!
